src/zarr/core/chunk_key_encodings.py

```diff
@@ -95,10 +80,38 @@ def encode_chunk_key(self, chunk_coords: tuple[int, ...]) -> str:
 @dataclass(frozen=True)
 class V2ChunkKeyEncoding(ChunkKeyEncoding):
     name: Literal["v2"] = "v2"
```

Comment:
Would annotating this as a `ClassVar` work here? Then we wouldn't need to redundantly pass a `name` parameter to `__init__`